VisualizationBase/src/Scene.cpp

```diff
@@ -150,16 +151,17 @@ ViewItem* Scene::currentViewItem()
 
 ViewItem* Scene::viewItem(const QString name)
 {
-	for (auto item : viewItems_)
-		if (item->name() == name)
-			return item;
+	for (auto vector : viewItems_)
+		for (auto item : vector)
+			if (item && item->name() == name)
+				return item;
 	return nullptr;
 }
 
 void Scene::switchToView(ViewItem *view)
 {
 	Q_ASSERT(!inAnUpdate_);
-	Q_ASSERT(viewItems_.contains(view));
+	Q_ASSERT(viewItems_[0].contains(view) || viewItems_[1].contains(view) || viewItems_[2].contains(view));
```

Comment:
as before do not hardcode the fact that there are 3. Use `std:any_of' or something similar.